input-profile.js

```diff
@@ -0,0 +1,176 @@
+                        this.getObjectByName(obj, visualResponses[j].maxNodeName);
+                    }
+                }
+            }
+        }
+    }
+
+    getObjectByName(obj, name) {
+        if (!obj || !name) return;
+        if (obj.name === name) this._gamepadObjects[name] = obj;
+        const children = obj.children;
+        for (let i = 0; i < children.length; ++i) this.getObjectByName(children[i], name);
+    }
+
+    assignTransform(target, min, max, value) {
+        vec3.lerp(
+            _TempVec,
+            min.getTranslationWorld([]),
+            max.getTranslationWorld([]),
+            value
+        );
+        target.setTranslationWorld(_TempVec);
+
+        quat.lerp(_TempQuat, min.rotationWorld, max.rotationWorld, value);
+        quat.normalize(_TempQuat, _TempQuat);
+        target.rotationWorld = _TempQuat;
+    }
+
+    mapGamepadInput() {
+        const components = this.ProfileJSON.layouts[this.handedness].components;
+        if (!components) return;
+        for (const i in components) {
+            if (components.hasOwnProperty(i)) {
+                const component = components[i];
+                const visualResponses = component.visualResponses;
+                for (const j in visualResponses) {
+                    if (visualResponses.hasOwnProperty) {
+                        const visualResponse = visualResponses[j];
+                        const target = this._gamepadObjects[visualResponse.valueNodeName];
+                        const min = this._gamepadObjects[visualResponse.minNodeName];
+                        const max = this._gamepadObjects[visualResponse.maxNodeName];
+
+                        this.assignTransform(
+                            target,
+                            min,
+                            max,
+                            this.getGamepadValue(component, visualResponse)
+                        );
+                    }
+                }
+            }
+        }
+    }
+
+    getGamepadValue(component, visualResponse) {
+        if (visualResponse.valueNodeProperty === 'transform') {
+            switch (component.type) {
+                case 'button':
+                    return this.gamepad.buttons[component.gamepadIndices.button].pressed;
+
+                case 'thumbstick':
+                    // check if component property matches with axis
+                    if (visualResponse.componentProperty === 'button') {
+                        return this.gamepad.buttons[component.gamepadIndices.button]
+                            .pressed;
+                    }
+                    if (visualResponse.componentProperty === 'xAxis') {
+                        return (this.gamepad.axes[component.gamepadIndices.xAxis] + 1) / 2;
+                    }
+                    if (visualResponse.componentProperty === 'yAxis') {
+                        return (this.gamepad.axes[component.gamepadIndices.yAxis] + 1) / 2;
+                    }
```

Comment:
Is it possible none of those `if` match? You are likely missing `break;` statement. If those if **must** match, then please use `elseif` or something that makes it easy understandable by reading the code that the next switch statement isn't going to be executed.